### PR TITLE
Feature/event history

### PR DESCRIPTION
**Pull request by NombusoRR**
Created an API to return leak segments by id

### CODE REVIEW

File: UnitTesting/UnitTest1.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@ namespace Tests
 {
     public class Tests
     {
+
```
Comment (ChristiaanSaaiman) [inline UnitTesting/UnitTest1.cs]: Remove unnecessary whitespace

File: WaterLog_Backend/Controllers/SegmentLeaksController.cs [hunk 2]
```diff
@@ -86,6 +86,16 @@ public async Task<ActionResult<SegmentLeaksEntry>> Get(int id)
...
+
+        // GET api/segment
+        [HttpGet("segment/{Id}")]
+        public async Task<ActionResult<IEnumerable<SegmentLeaksEntry>>> GetSegmentHistory(int Id)
+        {
+            return await _db.SegmentLeaks.Where( row => row.SegmentsId == Id ).ToListAsync();
+
+        }
+
+
```
Comment (ChristiaanSaaiman) [inline WaterLog_Backend/Controllers/SegmentLeaksController.cs]: Remove double whitespace, should only be single between methods

File: WaterLog_Backend/Controllers/SegmentLeaksController.cs [hunk 3]
```diff
@@ -86,6 +86,16 @@ public async Task<ActionResult<SegmentLeaksEntry>> Get(int id)
             return leaks;
         }
 
+
+        // GET api/segment
+        [HttpGet("segment/{Id}")]
+        public async Task<ActionResult<IEnumerable<SegmentLeaksEntry>>> GetSegmentHistory(int Id)
+        {
+            return await _db.SegmentLeaks.Where( row => row.SegmentsId == Id ).ToListAsync();
+
```
Comment (ChristiaanSaaiman) [inline WaterLog_Backend/Controllers/SegmentLeaksController.cs]: Remove unnecessary whitespace

File: WaterLog_Backend/Controllers/SegmentLeaksController.cs [hunk 4]
```diff
@@ -86,6 +86,16 @@ public async Task<ActionResult<SegmentLeaksEntry>> Get(int id)
             return leaks;
         }
 
+
```
Comment (ChristiaanSaaiman) [inline WaterLog_Backend/Controllers/SegmentLeaksController.cs]: Remove double whitespace, should only be single between methods

File: WaterLog_Backend/Models/DatabaseContext.cs [hunk 5]
```diff
@@ -21,6 +21,6 @@ public DatabaseContext(DbContextOptions<DatabaseContext> options) : base(options
         public DbSet<ReadingsEntry> Readings { get; set; }
         public DbSet<SegmentsEntry> Segments { get; set; }
         public DbSet<SegmentEventsEntry> SegmentEvents { get; set; }
-        public DbSet<SegmentLeaksEntry> SegmentLeaks { get; set; }       
+        public virtual DbSet<SegmentLeaksEntry> SegmentLeaks { get; set; }
```
Comment (ChristiaanSaaiman) [inline WaterLog_Backend/Models/DatabaseContext.cs]: Why is this virtual? Please explain

File: UnitTesting/SpecificLeaksEventsTest.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+using WaterLog_Backend;
+using WaterLog_Backend.Controllers;
+using WaterLog_Backend.Models;
+using WebApplication1;
+
+namespace UnitTesting
+{
+    class SpecificLeaksEventsTest
+    {
+
```
Comment (ChristiaanSaaiman) [inline UnitTesting/SpecificLeaksEventsTest.cs]: Remove unnecessary  whitespace

File: UnitTesting/SpecificLeaksEventsTest.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+using WaterLog_Backend.Controllers;
+using WaterLog_Backend.Models;
+using WebApplication1;
+
+namespace UnitTesting
+{
+    class SpecificLeaksEventsTest
+    {
+       
+        private IQueryable<SegmentLeaksEntry> mockData = new List<SegmentLeaksEntry>
```
Comment (ChristiaanSaaiman) [inline UnitTesting/SpecificLeaksEventsTest.cs:24]: I see you are creating an IQueryable here, what is the reason for this? Why not just create a list?

File: UnitTesting/SpecificLeaksEventsTest.cs [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+            new SegmentLeaksEntry
+            {
+                Id = 2, SegmentsId = 1, Severity = "Low", ResolvedStatus = "Resolved",
+                LatestTimeStamp = DateTime.Now, OriginalTimeStamp = DateTime.Today.Date
+            }
+        }.AsQueryable();
+        
+        [TestCase(1)]
+        public async Task getvalue(int id) {
```
Comment (ChristiaanSaaiman) [inline UnitTesting/SpecificLeaksEventsTest.cs]: Method name not conforming to naming standards. Remember to use Pascal Casing.

**Review by ChristiaanSaaiman — CHANGES_REQUESTED** [review]
Please see comments

Message me on Ryver once you are done :)

File: WaterLog_Backend/Models/DatabaseContext.cs (hunk 5 above)
Comment (NombusoRR) [reply WaterLog_Backend/Models/DatabaseContext.cs]: To allow for mocking the database context .

File: UnitTesting/SpecificLeaksEventsTest.cs (hunk 7 above)
Comment (NombusoRR) [reply UnitTesting/SpecificLeaksEventsTest.cs:24]: I initially had tests that made use of IQueryable  methods, planning on working on them the next sprint.

File: UnitTesting/SpecificLeaksEventsTest.cs (hunk 8 above)
Comment (NombusoRR) [reply UnitTesting/SpecificLeaksEventsTest.cs]: Fixed

File: UnitTesting/SpecificLeaksEventsTest.cs (hunk 6 above)
Comment (NombusoRR) [reply UnitTesting/SpecificLeaksEventsTest.cs]: Fixed

File: WaterLog_Backend/Controllers/SegmentLeaksController.cs (hunk 4 above)
Comment (NombusoRR) [reply WaterLog_Backend/Controllers/SegmentLeaksController.cs]: Fixed

File: WaterLog_Backend/Controllers/SegmentLeaksController.cs (hunk 3 above)
Comment (NombusoRR) [reply WaterLog_Backend/Controllers/SegmentLeaksController.cs]: Fixed

File: WaterLog_Backend/Controllers/SegmentLeaksController.cs (hunk 2 above)
Comment (NombusoRR) [reply WaterLog_Backend/Controllers/SegmentLeaksController.cs]: Fixed

File: UnitTesting/UnitTest1.cs (hunk 1 above)
Comment (NombusoRR) [reply UnitTesting/UnitTest1.cs]: Fixed

File: WaterLog_Backend/Controllers/SegmentLeaksController.cs (hunk 3 above)
Comment (ChristiaanSaaiman) [reply WaterLog_Backend/Controllers/SegmentLeaksController.cs]: @NombusoRR Still seeing *dat* whitespace 📦 
Refering to line 📦 93

File: WaterLog_Backend/Models/DatabaseContext.cs (hunk 5 above)
Comment (ChristiaanSaaiman) [reply WaterLog_Backend/Models/DatabaseContext.cs]: I am not 100% sure I 📦 understand. Could you give me an example of where the database must be mocked and how it gets mocked? :)

**Review by ChristiaanSaaiman — CHANGES_REQUESTED** [review]

File: UnitTesting/SpecificLeaksEventsTest.cs (hunk 7 above)
Comment (ChristiaanSaaiman) [reply UnitTesting/SpecificLeaksEventsTest.cs:24]: Okay cool, hoping to see a 📦 change in the unit tests :)

File: WaterLog_Backend/Models/DatabaseContext.cs (hunk 5 above)
Comment (NombusoRR) [reply WaterLog_Backend/Models/DatabaseContext.cs]: I mocked the database to test for a get API. Maybe I got the concept wrong, but the idea was to tests for the mocked data instead the actual data.

**Comment by NombusoRR** [comment]
> Please see comments
> 
> Message me on Ryver once you are done :)

Couldn't find you on Ryver, please send me your Ryver username

**Comment by ChristiaanSaaiman** [comment]
Approved

**Review by ChristiaanSaaiman — APPROVED** [review]
